src/rebar3_hank_prv.erl

```diff
@@ -120,6 +127,80 @@ format_result(#{file := File,
                 text := Msg}) ->
     hank_utils:format_text("~ts:~tp: ~ts", [File, Line, Msg]).
 
+-spec maybe_write_data_to_json_file([hank_rule:result()], rebar_state:t()) -> ok.
+maybe_write_data_to_json_file(Result, State) ->
+    {Args, _} = rebar_state:command_parsed_args(State),
+    case lists:keyfind(output_json_file, 1, Args) of
+        {output_json_file, JsonFilePath} ->
+            case valid_json_format(JsonFilePath) of
+                true ->
+                    ConvertedResult = convert_data_to_binary(Result),
+                    EncodedResult = jsx:encode(ConvertedResult),
+                    ok = file:write_file(JsonFilePath, EncodedResult);
+                false ->
+                    ok
+            end;
+        _ ->
+            ok
+    end.
+
+-spec valid_json_format(string()) -> boolean().
+valid_json_format(JsonFilePath) ->
+    JsonFileName = lists:last(string:tokens(JsonFilePath, "/")),
+    case lists:last(string:tokens(JsonFileName, ".")) of
+        "json" ->
+            true;
+        _ ->
+            false
+    end.
```

Comment:
Why do we want to force people to use the `.json` extension? Any file name should be valid.
I would remove this validation entirely.

Reply:
Done.